core/property/registry.go

```diff
@@ -2,31 +2,89 @@ package property
 
 import (
 	"fmt"
-	"sync"
-
-	"github.com/TIBCOSoftware/flogo-lib/core/data"
+	"github.com/TIBCOSoftware/flogo-lib/config"
 	"github.com/TIBCOSoftware/flogo-lib/logger"
+	"os"
+	"reflect"
+	"regexp"
+	"strings"
+	"sync"
 )
 
 var (
-	props = make(map[string]string)
+	props = make(map[string]interface{})
 	mut   = sync.RWMutex{}
+	regex = regexp.MustCompile(config.GetPropertyDelimiterFormat())
+	// Default Resolver
+	resolver Resolver = &DefaultResolver{}
 )
 
+// Resolve value sourced from Enviornment variable or any other configuration management services
+type Resolver interface {
+	//Resolve value for given name
+	Resolve(name string) interface{}
+	CanResolve(name string) bool
+}
+
+type DefaultResolver struct {
+}
+
+// Default resolver resolves values from property bag and environment variable
+func (resolver *DefaultResolver) Resolve(value string) interface{} {
+	if len(value) == 0 {
+		return value
+	}
+
+	// Value format: ${env.ENVVAR1}
+	if strings.Contains(value, "${env.") {
+		value = value[6 : len(value)-1]
+		logger.Debugf("Resolving  value for enviornment variable: '%s'", value)
+		return os.Getenv(value)
+	} else if strings.Contains(value, "${property.") {
+		// Value format: ${property.Prop1}
+		property := value[11 : len(value)-1]
+		logger.Debugf("Resolving  value for property : '%s'", property)
+		return Get(property)
+	}
+
+	return value
+}
+
+func (resolver *DefaultResolver) CanResolve(value string) bool {
+	return regex.MatchString(value)
+}
+
 // Get returns the value of the property for the given id
-// If it is an environment property (for example {MY_PROP})
-// The value will be looked up in the os environment
-func Get(id string) string {
+func Get(id string) interface{} {
 	mut.RLock()
 	defer mut.RUnlock()
 	prop, ok := props[id]
+
 	if !ok {
+		if resolver.CanResolve(id) {
```

Comment:
For the default resolver we decided to not having nested resolution, we will live that for contributed resolvers